gammapy/utils/pbar.py

```diff
@@ -0,0 +1,36 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Utilities for progress bar display"""
+import logging
+from collections.abc import Iterable
+
+log = logging.getLogger(__name__)
+
+try:
+    from tqdm.auto import tqdm
+except ImportError:
+    class tqdm():
+        def __init__(self, iterable, disable=True, **kwargs):
+            self.disable = disable
+            self._iterable = iterable
+            if self.disable == False:
+                log.info(
+                    f"Tqdm is currently not installed. Visit https://tqdm.github.io/"
+                )
+
+        def update(self, x):
+            pass
+
+        def __iter__(self):
+            return self._iterable.__iter__()
+
+        def __next__(self):
+            return self._iterable.__next__()
+
+def progress_bar(iterable, show_progress_bar=False, desc=None):
+    if not isinstance(iterable, Iterable) and show_progress_bar == True:
+        raise AttributeError("Can't set up the progress bar if total is None")
```

Comment:
I think the correct error here is `ValueError`, how does `tqdm` fail if users pass and non-iterable object?

Reply:
`TypeError: 'int' object is not iterable`

Maybe I could even remove these three lines (29-30-31), because `tqdm` will take care of it already?